src/main/java/com/yahoo/bullet/querying/partitioning/SimpleEqualityPartitioner.java

```diff
@@ -0,0 +1,210 @@
+    public static final char FALSE_CHAR = '0';
+
+    private List<String> fields;
+    private Set<String> fieldSet;
+    private String delimiter;
+    private final List<String> defaultKeys;
+
+    /**
+     * Constructor that takes a {@link BulletConfig} instance with definitions for the various settings this needs.
+     * Delimiter: {@link BulletConfig#EQUALITY_PARTITIONER_DELIMITER} and
+     * Fields to partition on: {@link BulletConfig#EQUALITY_PARTITIONER_FIELDS}
+     *
+     * @param config The non-null config containing settings for this class.
+     */
+    public SimpleEqualityPartitioner(BulletConfig config) {
+        delimiter = config.getAs(BulletConfig.EQUALITY_PARTITIONER_DELIMITER, String.class);
+        fields = (List<String>) config.getAs(BulletConfig.EQUALITY_PARTITIONER_FIELDS, List.class);
+        fieldSet = new HashSet<>(fields);
+        String defaultKey = Collections.nCopies(fields.size(), NO_FIELD).stream().collect(Collectors.joining(delimiter));
+        defaultKeys = Collections.singletonList(defaultKey);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * This partitioner ensures that queries are not stored in duplicate by returning only key for a query (the list
+     * that is returned is of size 1).
+     *
+     * @param query The query to partition.
+     * @return The {@link List} containing the one key for this query.
+     */
+    @Override
+    public List<String> getKeys(Query query) {
+        Objects.requireNonNull(query);
+        List<Clause> filters = query.getFilters();
+        // If no filters or has non ANDs, default partition
+        if (filters == null || filters.isEmpty() || hasNonANDLogicals(filters)) {
+            return defaultKeys;
+        }
+
+        // For each unique field, get all the FilterClauses that define an operation on it.
+        Map<String, List<FilterClause>> fieldFilters = new HashMap<>();
+        filters.forEach(c -> this.mapFieldToFilters(c, fieldFilters));
+
+        // If not one equality filter per field and not one value per filter, default partition
+        if (fieldFilters.values().stream().anyMatch(this::hasInvalidFilterClauses)) {
+            return defaultKeys;
+        }
+
+        // Generate key in fields order and pad with NO_FIELD if no mapping present
+        String key = fields.stream().map(fieldFilters::get).map(this::getFilterValue).collect(Collectors.joining(delimiter));
+        // For the SimpleEqualityPartitioner, the query is mapped to exactly one key only.
+        return Collections.singletonList(key);
+    }
+
+    @Override
+    public List<String> getKeys(BulletRecord record) {
+        Map<String, String> values = getFieldValues(record);
+        // For fields not present (NO_FIELD mapped), this de-dupes them with the binary combination that ignores them
+        Set<String> keys = new HashSet<>();
+
+        int numberOfFields = fields.size();
+        int combinations = 1 << numberOfFields;
```

Comment:
I don't want to set a maximum because there can be queries and records in an instance of bullet where having a lot of fields to partition on might make sense. I'd rather leave it up to the person launching Bullet to set fields to not have that many.

Reply:
how about limit it to 30 then? 2^31 is MIN_INT

Reply:
Used 10 as the limit